src/hooks/amm.ts

```diff
@@ -0,0 +1,95 @@
+import { Provider } from "@ethersproject/providers";
+import { Token } from "@token";
+import { BigNumberish, ethers, Signer } from "ethers";
+import { useSignerFromEthereum } from "./signer";
+import SWAP_ABI from "@abis/swap.json";
+import {
+    contractsForChainId,
+  getTimeMinutesFromNow,
+  uiToNative,
+} from "@common/utils";
+
+/**
+ * @notice Call the `swap` function of the `Swap` contract
+ * @param poolName the name of the pool to use
+ * @param chainId the id of the chain to swap on
+ * @param signer the signer who is calling the swap
+ * @param tokenFrom the token to swap from
+ * @param tokenTo the token to swap to
+ * @param amountFrom the amount of tokenFrom to swap
+ * @param amountTo optional: the minimum amount of tokenTo to receive
+ * @param deadline optional: the deadline for the txn to execute
+ * @returns amount of tokenTo received from the swap
+ */
+export async function swap(
```

Comment:
should this execute a swap, or build a txn and return the txn?

Reply:
Let's separate out into two functions as discussed earlier today: 
-  One function that builds an unsigned / populated transaction e.g `return await swapContract.populateTransaction.swap`
- One function that utilizes the build tx to then execute signing on behalf of the caller, assuming a signing provider is attached